### PR TITLE
Replace all exported method return types with structures

### PR DESCRIPTION
**Pull request by anorth**
This is part of getting the exports to have a consistent type signature for method dispatch. These structures should be cbor-encoded for on-chain representation in a follow-up.

[We have already added params structs for the mutlisig actor, other actors are still to come]

### CODE REVIEW

File: actors/builtin/init/init_actor.go [hunk 1]
```diff
@@ -54,10 +52,15 @@ func (a *InitActor) Constructor(rt Runtime) InvocOutput {
...
 	}
 	UpdateRelease(rt, h, st)
-	return InvocOutput{}
+	return &vmr.EmptyReturn{}
 }
 
-func (a *InitActor) Exec(rt Runtime, execCodeID abi.ActorCodeID, constructorParams abi.MethodParams) InvocOutput {
+type ExecReturn struct {
+	IDAddress     addr.Address // The canonical ID-based address for the actor.
+	RobustAddress addr.Address // A mre expensive but re-org-safe address for the newly created actor.
```
[the remark this answers is not in the document]
Comment (frrist) [reply actors/builtin/init/init_actor.go]: never mind you meant 'more'

File: actors/builtin/reward/reward_actor.go [hunk 2]
```diff
@@ -12,7 +12,6 @@ import (
 	cid "github.com/ipfs/go-cid"
 )
 
-type InvocOutput = vmr.InvocOutput
 type Runtime = vmr.Runtime
```
Comment (frrist) [inline actors/builtin/reward/reward_actor.go]: wanna nix this while you're at it?

File: actors/builtin/reward/reward_actor.go [hunk 3]
```diff
@@ -12,7 +12,6 @@ import (
...
 
 var TODO = autil.TODO
```
Comment (frrist) [inline actors/builtin/reward/reward_actor.go]: and this

File: actors/builtin/storage_market/storage_market_actor_boilerplate.go [hunk 4]
```diff
@@ -17,7 +17,6 @@ type DealIDQueue = autil.DealIDQueue
 // workaround due to the lack of generics support in Go.
 ////////////////////////////////////////////////////////////////////////////////
 
-type InvocOutput = vmr.InvocOutput
 type Runtime = vmr.Runtime
```
Comment (frrist) [inline actors/builtin/storage_market/storage_market_actor_boilerplate.go]: can we remove all of this?

File: actors/runtime/runtime.go [hunk 5]
```diff
@@ -106,17 +109,20 @@ type Syscalls interface {
 	VerifyPoSt(sectorSize abi.SectorSize, vi abi.PoStVerifyInfo) bool
 }
 
+// The return type from a message send from one actor to another. This abstracts over the internal representation of
+// the return, in particular whether it has been serialized to bytes or just passed through.
+// Production code is expected to de/serialize, but test and other code may pass the value straight through.
```
Comment (frrist) [inline actors/runtime/runtime.go]: :pray: nice

File: actors/builtin/storage_market/storage_market_actor_boilerplate.go (hunk 4 above)
Comment (anorth) [reply actors/builtin/storage_market/storage_market_actor_boilerplate.go]: We'll get there incrementally.

**Pull request merged**